integration-tests/api/factories/simple-product-collection-factory.ts

```diff
@@ -0,0 +1,39 @@
+import { Connection } from "typeorm"
+import faker from "faker"
+import { ProductCollection } from "@medusajs/medusa"
+
+export type Data = {
+  title?: string,
+  handle: string
+}
+
+export const simpleProductCollectionFactory = async <
+  TData extends Data | Data[] = Data | Data[],
+  TResult = TData extends Array<Data> ? ProductCollection[] : ProductCollection
+>(
+  connection: Connection,
+  data: TData,
+  seed?: number
+): Promise<TResult> => {
+  if (typeof seed !== "undefined") {
+    faker.seed(seed)
+  }
+
+  const manager = connection.manager
+
+  const collectionsData = Array.isArray(data) ? data : [data]
+
+  const collections: ProductCollection[] = []
+
+  for (const collectionData of collectionsData) {
+    const collection_ = manager.create(ProductCollection, {
+      id: `simple-id-${Math.random() * 1000}`,
+      title: collectionData.title ?? faker.datatype.string(10),
+      handle: collectionData.handle
```

Comment:
comment: Sorry, I actually meant to use it for `title` and omit handle as we have a `beforeInsert` trigger, that kebab-cases the title. That way, we don't have to require any data for the factory.

But really, you don't have to include it. It's nitpicking 😅

Reply:
I can make the handle optional but i will still have to pass some data to the factory as I am dependent from the data in the csv template for the tests ^^

Reply:
I think we can merge as it is for now wdyt?

Reply:
Yep!

Reply:
The problem I see with using the title is that you can have typos...  ie: in our use case, we had a csv, and some collection title had for example "A Collection Title", and some other had "A-Collection Title"  and another "A Collections Title" but the handle is the same...  It's easier not to messup with fewer products, but in our case there are 32k products...  Not the end of the world tough, we can go back and manually reassign.

Reply:
@RegisHubelia this is a factory used for the integration tests and it is not related to the strategy implementation. This is also why the strategy is based on the handle and not the title 💪

Reply:
@olivermrbl I have included it in the end :D